### PR TITLE
refactor(console,core,phrases): replace `LanguageKey` with `LangaugeTag` in phrase-ui related code

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- refactor(console,core,phrases): replace `LanguageKey` with `LangaugeTag` in phrase-ui related code
- add `getDefaultLanguageTag` to `phrases-ui` package, because this logic should not shared with `phrases` package by the `@logto/core-kit`.
- add `isBuiltInLanguageTag` util to `phrases-ui` package because the `built-in` is only related to the phrases-ui translations.
- add new types in `phrases-ui` package for better type resolution.

## Testing

Test locally and UT passed.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
   },
   "dependencies": {
     "@logto/connector-kit": "^1.0.0-beta.13",
```
Comment (IceHe) [inline packages/core/package.json:25]: better align `*-kit` version? (connector-kit also depends on language-kit)

**Review by charIeszhao — APPROVED** [review]

File: packages/phrases-ui/src/types.ts [hunk 2]
```diff
@@ -1,19 +1,3 @@
-import { LanguageKey, languageKeyGuard } from '@logto/core-kit';
-
 import en from './locales/en';
 
 export type LocalePhrase = typeof en;
```
Comment (IceHe) [inline packages/phrases-ui/src/types.ts:3]: maybe also move to index.ts?

File: packages/console/src/pages/SignInExperience/components/ManageLanguageModal/index.tsx [hunk 3]
```diff
@@ -26,31 +26,30 @@ const ManageLanguageModal = ({ isOpen, onClose }: ManageLanguageModalProps) => {
...
+    const allTags = customUiLanguageTags?.length
+      ? [...new Set([...builtInUiLanguages, ...customUiLanguageTags])]
+      : builtInUiLanguages;
 
-    return allKeys.slice().sort();
+    return allTags.slice().sort();
   }, [customPhraseResponses]);
 
-  const defaultLanguageKey = getDefaultLanguage(allLanguageKeys[0] ?? '');
+  const defaultLanguageTag = getDefaultLanguageTag(allLanguageTags[0] ?? '');
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/components/ManageLanguageModal/index.tsx:39]: assume that the first custom phrase language key is zh-TW (`allLanguageTags[0] === 'zh-TW'`),
and `getDefaultLanguageTag(allLanguageTags[0] ?? '')` will be `en`.
Is it expected?

File: packages/phrases-ui/src/types.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/phrases-ui/src/types.ts:3]: This type is used in locale files, we cannot move it to the index file.

File: packages/console/src/pages/SignInExperience/components/ManageLanguageModal/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/components/ManageLanguageModal/index.tsx]: updated.

File: packages/core/package.json (hunk 1 above)
Comment (xiaoyijun) [reply packages/core/package.json:25]: No updates related to the `@logto/connector-kit`, update it when needed.

**Review by IceHe — APPROVED** [review]

File: packages/phrases-ui/src/index.ts [hunk 4]
```diff
@@ -23,4 +37,10 @@ const resource: Resource = {
   'tr-TR': trTR,
 };
 
+export const getDefaultLanguageTag = (language: string): LanguageTag =>
+  builtInLanguageTagGuard.or(fallback<LanguageTag>('en')).parse(language);
```
Comment (IceHe) [inline packages/phrases-ui/src/index.ts:41]:
```suggestion
export const getDefaultLanguageTag = (language: unknown): LanguageTag =>
  builtInLanguageTagGuard.or(fallback<LanguageTag>('en')).parse(language);
```

Comment (xiaoyijun) [reply packages/phrases-ui/src/index.ts:41]: `string` is safer here.

**Pull request merged**